### DISCUSSION TITLE
Python-Gitlab - Creating a subrepo and forking a group over to new sub repo using the API

### DISCUSSION THREAD

**Discussion by zechariahmoore** [python-gitlab/python-gitlab#1984] [Q&A]
hHey all I'm trying to demonstrate some code (python). Eventually this will be automated but we're just trying to trigger off the gitlab-python api right now. 

The first of this is code is to create a new subgroup which uses a variable value to create a subgroup name. The variable value will be directly related to the client ID and name. 

client_id length: 10 char

name length: 15 char

After having the client subgroup created - our second goal is to fork a existing repo into this subgroup. This will be automated as well. 

I have my base code here - but it needs some love. Any pointers of how you'd put this together? 

```
import gitlab

# authenticate
token = 'get-yo-own-key'

# def list groups
def list_groups():
    #  lists existing visible groups
    glist = gl.groups.list()
    return glist

# def create subgroup
def create_subgroup(sub_group_name, groupid):
    #  creates a subgroup
    sub_group = gl.groups.create({"name": sub_group_name, "path": sub_group_name, "parent_id": groupid})
    return sub_group

# def list projects
def list_projects():
    #  lists existing visible groups
    plist = gl.projects.list(owned=True)
    return plist

# def fork repo
def fork_repo(sub_group_id, projectid):
    #  creates fork in subgroup
    project = gl.projects.get(projectid)
    print(project)
    fork = project.forks.create({"namespace_id": sub_group_id})
    print(fork)


#gl = gitlab.Gitlab(private_token=token)
glist = list_groups()
for gobj in glist:
     print(gobj)

# print new subgroup to get ID
group = gl.groups.get("new_subgroup_id")
print(group)

# list project
plist = list_projects()
for pobj in plist:
    print(pobj)

# fork repo 
fork_repo("new_subgroup_ID", "repo_pull_id")
```

Other notes: We would like a way to get a specific group_id using the name  (this may be confusing if we have similarly named groups). I have the below as a starting point thrown together 

```#  authenticate
gl = gitlab.Gitlab('https://gitlab.com/', 'get yo own key')

# set gitlab group id
group_id = 12929416 
group = gl.groups.get(group_id, lazy=True)


# get all groups
groups = gl.groups.list(include_subgroups=False, all=True)

# get all group ids
group_ids = []
for group in groups:
    group_ids.append(group.name, group,id)

print(group_ids)
```

this obviously isn't done as I'm rehashing it from scrip that gets all project ids for a gitlab group - but I'd like to be able to get a specific group ID by name.

Any input is welcome - sorry I'm a bit new to all this.

**Comment by JohnVillalovos** [comment]
@zechariahmoore  I'm having a hard time to understand what is the question. Can you add a 1-2 line question? Or a list of questions.

As far as doing a fork. There is an example at: https://python-gitlab.readthedocs.io/en/main/gl_objects/projects.html#examples and upstream docs at: https://docs.gitlab.com/ee/api/projects.html#fork-project

**Reply by walterrowe** [reply]
do they mean submodule vs subrepo?
